src/main/webapp/js/finscholar/commonlistview/commonlistview.js

```diff
@@ -0,0 +1,88 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** @fileoverview The mini controller for scholarship list view. */
+
+goog.module('finscholar.commonlistview');
+
+const {ScholarshipListDataHandler} = goog.require('datahandlers.scholarshiplistdatahandler');
+const {commonlistview, scholarshiplistitems, collegelistitems} = goog.require('finscholar.commonlistview.templates');
+const googDom = goog.require('goog.dom');
+
+const ITEM_CONTAINER_ID = 'table-body';
+
+/** The mini controller for scholarship list view. */
+class CommonListView {
+  
+  constructor(dataHandler, optionIndex) {
+    /** @private @const {ScholarshipListDataHandler | CollegeListDataHandler} */
+    this.dataHandler_ = dataHandler;
+    /** @private @const {string} */
+    this.optionIndex_ = optionIndex;
+    /** @private @const {function({scholarships : !Array<?>}):Element} */
+    this.template_ = this.optionIndex_ == '0' ? collegelistitems : scholarshiplistitems;
+    /** @private {number} The number of batch of data has been loaded into the view. */
+    this.batch_ = 0;
+    /** @private {Element|null} The container for all list items. */
+    this.container_ = null;
+    /** @private @const {!function():Promise<undefined>} */
+    this.bindedScrollHandler_ = this.loadNextBatch_.bind(this);
+    /** @private @const {!function(number):Promise<undefined>} */
+    this.bindedScholarshipLoader_ = this.renderNextBatch_.bind(this);
+    /** @private {number} */
+    this.itemsPerBatch_ = 15;
```

Comment:
Is this mutable? Just wondering cuz the colleges API defaults to 20 entries per page, I think I should be able to request 15 at a time though if need be.

Reply:
Yeah I'll implement a method to set that (although I don't know which part of the app will use the setter).
Actually I can just change the value to 20 lol